osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -290,6 +293,7 @@ private void load(AudioManager audio, OsuColour colours, ManageCollectionsDialog
                 BeatmapOptions.AddButton(@"Delete", @"all difficulties", FontAwesome.Solid.Trash, colours.Pink, () => delete(Beatmap.Value.BeatmapSetInfo));
                 BeatmapOptions.AddButton(@"Remove", @"from unplayed", FontAwesome.Regular.TimesCircle, colours.Purple, null);
                 BeatmapOptions.AddButton(@"Clear", @"local scores", FontAwesome.Solid.Eraser, colours.Purple, () => clearScores(Beatmap.Value.BeatmapInfo));
+                BeatmapOptions.InsertButton(2, @"View", @"details", FontAwesome.Solid.Bars, colours.Blue, () => beatmapOverlay?.FetchAndShowBeatmap(Beatmap.Value.BeatmapInfo.OnlineID));
```

Comment:
Why is this required? Why can't you just move it up?

Reply:
Move it up in what way?

Reply:
This is required because inheritors of `SongSelect` also add their own buttons (e.g. `PlaySongSelect` adding "Edit Beatmap"), but this button wants to be ordered last, so it uses a depth of its own that makes it placed last.

This can work alternative by just adding a virtual method for inheritors to override and add buttons, then place the view details button afterwards, i.e.:
```csharp
                ...
                BeatmapOptions.AddButton(@"Remove", @"from unplayed", FontAwesome.Regular.TimesCircle, colours.Purple, null);
                BeatmapOptions.AddButton(@"Clear", @"local scores", FontAwesome.Solid.Eraser, colours.Purple, () => clearScores(Beatmap.Value.BeatmapInfo));
                
                AddButtons();
                
                BeatmapOptions.AddButton(@"View", @"details", FontAwesome.Solid.Bars, colours.Blue, () => beatmapOverlay?.FetchAndShowBeatmap(Beatmap.Value.BeatmapInfo.OnlineID));

Reply:
Using a virtual method for this could work, but it would only allow buttons to be inserted into a specific location instead of an arbitrary one. It could definitely be enough though, and it's definitely another option

Reply:
Does "the proposed solution" refer to the virtual method? If so I'll implement that asap.

Reply:
Yes.